pkg/kubelet/container/runtime.go

```diff
@@ -58,6 +58,9 @@ type Runtime interface {
 	// APIVersion returns the API version information of the container
 	// runtime. This may be different from the runtime engine's version.
 	APIVersion() (Version, error)
+	// CheckVersionCompatibility returns whether the version of underlying runtime is compatible with
+	// kubelet's minimum requirement.
+	CheckVersionCompatibility() error
```

Comment:
Could this be `Status()` instead? We can evolve that to include other checks in the feature if necessary. Kubelet main doesn't care about the specific kinds of failures.